wordpress-mobile/WordPress-Aztec-iOS
# Applying styles at cursor position

### Details:
This PR adds support to toggle Text Styles at the cursor's position, without requiring the user to preselect the paragraphs, previously.

Needs Review: @diegoreymendez 
Closes #167

### Testing: No Selection
1. Hit the `Bold`/ `Italic` / `Underline` / `Strikethrough` button
2. Verify that the Editor behaves as expected

### Testing: Selection
1. Select a snippet of text
2. Hit the `Bold`/ `Italic` / `Underline` / `Strikethrough` button
3. Verify that the selected snippet gets styled as expected

### Testing: Toolbar
1. Place the cursor over text in `Bold`/ `Italic` / `Underline` / `Strikethrough`
2. Verify that the Toolbar lights as expected

### Testing: Paste
1. Toggle any of the `Bold`/ `Italic` / `Underline` / `Strikethrough` styles
2. Paste some random text
3. Verify that the pasted text reflects the style that was previously selected.

**Note:** This behavior is different from the one outlined in the original issue. Reasons why:
- Pages does it this way ("paste and match style")
- If you explicitly select `Bold` / else, and proceed to paste, it makes sense to respect that preselected style

### Testing: Selection
1. Toggle any of the text styles
2. Select a random snippet of text
3. Type anything

Verify that the styles selected in step (1) get lost
